src/core/js/models/adaptModel.js

```diff
@@ -306,8 +308,80 @@ define(function (require) {
 
         setOptional: function(value) {
             this.set({_isOptional: value});
-        }
+        },
+
+        checkLocking: function() {
+            switch (this.get("_lockType")) {
+                case "sequential":
+                    this.setSequentialLocking();
+                    break;
+                case "unlockFirst":
+                    this.setUnlockFirstLocking();
+                    break;
+                case "lockLast":
+                    this.setLockLastLocking();
+                    break;
+                case "custom":
+                    this.setCustomLocking();
```

Comment:
could you add a `default` case here that does something like:
`console.warn("AdaptModel.checkLocking: unknown _lockType of '" + this.get("_lockType") + "' found on " + this.get('_id'));`